tests/components/roborock/conftest.py

```diff
@@ -95,82 +100,171 @@ def bypass_api_client_fixture() -> None:
-        ),
-        patch(
-            "homeassistant.components.roborock.RoborockMqttClientV1.get_networking",
-            return_value=NETWORK_INFO,
-        ),
-        patch(
-            "homeassistant.components.roborock.coordinator.RoborockLocalClientV1.get_prop",
-            return_value=PROP,
-        ),
-        patch(
-            "homeassistant.components.roborock.coordinator.RoborockMqttClientV1.get_multi_maps_list",
-            return_value=MULTI_MAP_LIST,
-        ),
-        patch(
-            "homeassistant.components.roborock.coordinator.RoborockLocalClientV1.get_multi_maps_list",
-            return_value=MULTI_MAP_LIST,
-        ),
-        patch(
-            "homeassistant.components.roborock.coordinator.RoborockMapDataParser.parse",
-            return_value=MAP_DATA,
-        ),
-        patch(
-            "homeassistant.components.roborock.coordinator.RoborockLocalClientV1._send_message"
-        ),
-        patch("homeassistant.components.roborock.RoborockMqttClientV1._wait_response"),
-        patch(
-            "homeassistant.components.roborock.coordinator.RoborockLocalClientV1._wait_response"
-        ),
-        patch(
-            "roborock.version_1_apis.AttributeCache.async_value",
-        ),
-        patch(
-            "roborock.version_1_apis.AttributeCache.value",
-        ),
-        patch(
-            "homeassistant.components.roborock.coordinator.MAP_SLEEP",
-            0,
-        ),
-        patch(
-            "homeassistant.components.roborock.coordinator.RoborockLocalClientV1.get_room_mapping",
-            return_value=[
-                RoomMapping(16, "2362048"),
-                RoomMapping(17, "2362044"),
-                RoomMapping(18, "2362041"),
-            ],
-        ),
-        patch(
-            "homeassistant.components.roborock.coordinator.RoborockMqttClientV1.get_room_mapping",
-            return_value=[
-                RoomMapping(16, "2362048"),
-                RoomMapping(17, "2362044"),
-                RoomMapping(18, "2362041"),
+class FakeDevice(RoborockDevice):
+    """A fake device that returns a list of devices."""
+
+    is_connected: bool = True
+    is_local_connected: bool = True
+
+    def __init__(
+        self,
+        device_info: HomeDataDevice,
+        product: HomeDataProduct,
+    ) -> None:
+        """Initialize the FakeDevice."""
+        super().__init__(device_info, product, Mock(), Mock())
+
+    async def close(self) -> None:
+        """Close the device."""
+
+
+class FakeDeviceManager:
+    """A fake device manager that returns a list of devices."""
+
+    def __init__(self, devices: list[RoborockDevice]) -> None:
+        """Initialize the fake device manager."""
+        self._devices = devices
+
+    async def get_devices(self) -> list[RoborockDevice]:
+        """Return the list of devices."""
+        return self._devices
+
+
+def make_fake_switch(obj: Any) -> Any:
```

Comment:
I don't like the Any here and in the other functions, as it makes it hard for others to know which functions/attributes the object should have